dotnet/src/IntegrationTests/Planners/Stepwise/FunctionCallingStepwisePlannerTests.cs

```diff
@@ -104,8 +104,8 @@ public async Task ThrowsWhenPluginFunctionThrowsCriticalExceptionAsync()
             new FunctionCallingStepwisePlannerConfig() { MaxIterations = 5 });
 
         // Act & Assert
-        // Planner should call ThrowingEmailPluginFake.GetEmailAddressAsync, which throws InvalidProgramException
```

Comment:
Have you considered only adding the WriteJokeAsync function as a plugin to ensure that it doesn't get confused call the wrong one? Something like:

```
kernel.Plugins.Add(
     KernelPluginFactory.CreateFromFunctions(
        "JokePlanner",
        new[] {
            KernelFunctionFactory.CreateFromMethod({ThrowingEmailPluginFake instance}.WriteJokeAsync)
        }));
```

Reply:
I didn't realize this was a supported pattern, but I agree it makes sense here. Updated!